### PR TITLE
All members should ack before becoming active...

### PR DESCRIPTION
**Pull request by jumaffre**
... even when the service is `OPENING`.

The service checks that on opening the number of active members is `>=` than the recovery threshold (otherwise, we would be opening a service that is not recoverable). Consequently, recovery shares are now only issued when the service is opened.

Resolves https://github.com/microsoft/CCF/issues/1113

### CODE REVIEW

File: src/node/genesis_gen.h [hunk 1]
```diff
@@ -148,6 +131,29 @@ namespace ccf
...
+        throw std::logic_error(fmt::format(
+          "Member {} cannot be activated as they do not exist", member_id));
+      }
+
+      if (member->status == MemberStatus::ACCEPTED)
+      {
+        member->status = MemberStatus::ACTIVE;
+        if (get_active_members_count() >= max_active_members_count)
+        {
+          throw std::logic_error(fmt::format(
```
Comment (ashamis) [inline src/node/genesis_gen.h:149]: consider using cc_logic_error, this will also produce a stack trace when running in virtual

Comment (achamayou) [reply src/node/genesis_gen.h:149]: @ashamis should we just do a pass and replace'em all?

File: src/node/rpc/test/member_voting_test.cpp [hunk 2]
```diff
@@ -1777,78 +1797,113 @@ DOCTEST_TEST_CASE("Maximum number of active members")
...
+        crypto::BoxKey::public_from_private(private_encryption_key));
+      auto id = gen.add_member(cert, public_encryption_key.raw());
+      members[id] = {cert, private_encryption_key};
     }
+    gen.set_recovery_threshold(recovery_threshold);
     gen.finalize();
+    frontend.open();
+  }
+
+  DOCTEST_INFO("Opens fails as recovery threshold is too high");
```
Comment (achamayou) [inline src/node/rpc/test/member_voting_test.cpp:1880]:
```suggestion
  DOCTEST_INFO("Open fails as recovery threshold is too high");
```

File: src/node/genesis_gen.h (hunk 1 above)
Comment (jumaffre) [reply src/node/genesis_gen.h:148]: Raised https://github.com/microsoft/CCF/issues/1324 as it is something we should do consistently in our codebase.

**Pull request merged**